src/components/network-map-tab.js

```diff
@@ -194,78 +183,123 @@ export const NetworkMapTab = ({
+                    const newGeoData = new GeoData(
+                        positions[0].length > 0
+                            ? new Map(geoData.substationPositionsById)
+                            : geoData.substationPositionsById,
+                        positions[1].length > 0
+                            ? new Map(geoData.linePositionsById)
+                            : geoData.linePositionsById
+                    );
+                    setGeoData(newGeoData);
+                    setWaitingLoadGeoData(false);
+                });
+            } else {
+                console.info(`Loading geo data of study '${studyUuid}'...`);
+                setWaitingLoadGeoData(true);
+                const substationPositions = fetchSubstationPositions(
+                    studyUuid,
+                    currentNode?.id
+                );
+
+                const linePositions = lineFullPath
+                    ? fetchLinePositions(studyUuid, currentNode?.id)
+                    : [];
+
+                Promise.all([substationPositions, linePositions])
+                    .then((values) => {
+                        const newGeoData = new GeoData();
+                        newGeoData.setSubstationPositions(values[0]);
+                        newGeoData.setLinePositions(values[1]);
+                        setGeoData(newGeoData);
+                        setWaitingLoadGeoData(false);
+                    })
+                    .catch(function (error) {
+                        console.error(error.message);
+                        setWaitingLoadGeoData(false);
+                        setErrorMessage(
+                            intlRef.current.formatMessage(
+                                { id: 'geoDataLoadingFail' },
+                                { studyUuid: studyUuid }
+                            )
+                        );
+                    });
+            }
+        }
     }, [
-        currentNode?.id,
-        dispatch,
+        currentNode,
         intlRef,
         lineFullPath,
         setErrorMessage,
         studyUuid,
+        getMissingEquipmentsPositions,
+        network,
+        geoData,
     ]);
 
-    const handleReloadMap = useCallback(() => {
-        reloadMapGeoData();
-        dispatch(setForceNetworkReload());
-    }, [dispatch, reloadMapGeoData]);
+    const reloadMapGeoDataRef = useRef();
+    reloadMapGeoDataRef.current = reloadMapGeoData;
 
     useEffect(() => {
-        let previousCurrentNode = currentNodeRef.current;
-        currentNodeRef.current = currentNode;
-        // if only renaming, do not reload geo data
-        if (isNodeRenamed(previousCurrentNode, currentNode)) return;
-        if (disabled) return;
-        if (mapManualRefresh && isInitialized) return;
-        // Hack to avoid reload Geo Data when switching display mode to TREE then back to MAP or HYBRID
-        // TODO REMOVE LATER
-        if (!reloadMapNeeded) return;
+        reloadMapGeoDataRef.current();
```

Comment:
De we want to do it in mapManualRefresh Mode ?

Reply:
I think yes. Even if it does not really have sense